x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -122,6 +123,7 @@ protected static Batch<LogicalPlan> substitutions() {
             "Substitutions",
             Limiter.ONCE,
             new ReplaceLookupWithJoin(),
+            new SubstituteFilteredExpression(),
```

Comment:
I think it's important to comment why this is needed: this has to run before `ReplaceStatsNestedExpressionWithEval` and `ReplaceStatsAggExpressionWithEval` can be applied.

Reply:
@costin maybe this was overlooked due to having a thumbs-up from Bogdan already :D A small comment would be great - either here or on `SubstituteFilteredExpression`.